operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.logging.LogRecord;
+
+import com.meterware.simplestub.Memento;
+import com.meterware.simplestub.StaticStubSupport;
+import io.kubernetes.client.openapi.models.V1ResourceRule;
+import io.kubernetes.client.openapi.models.V1SelfSubjectRulesReview;
+import io.kubernetes.client.openapi.models.V1SubjectRulesReviewStatus;
+import oracle.kubernetes.operator.helpers.AuthorizationProxy.Operation;
+import oracle.kubernetes.operator.helpers.AuthorizationProxy.Resource;
+import oracle.kubernetes.operator.helpers.CrdHelper;
+import oracle.kubernetes.operator.helpers.KubernetesTestSupport;
+import oracle.kubernetes.operator.helpers.KubernetesVersion;
+import oracle.kubernetes.operator.helpers.TuningParametersStub;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.operator.work.TerminalStep;
+import oracle.kubernetes.utils.TestUtils;
+import org.hamcrest.Description;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static oracle.kubernetes.operator.StartupControlTest.StepPresenceMatcher.containsStep;
+import static oracle.kubernetes.operator.logging.MessageKeys.CRD_NO_READ_ACCESS;
+import static oracle.kubernetes.operator.logging.MessageKeys.NS_NO_READ_ACCESS;
+import static oracle.kubernetes.utils.LogMatcher.containsWarning;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+public class StartupControlTest {
+  private static final KubernetesVersion VERSION = new KubernetesVersion(1, 8);
```

Comment:
1,8? seems old, should we update?

Reply:
We absolutely should. There is still logic that determines whether Rules Review is supported, even though we no longer support versions before 1.8. I don't think it should be done as part of this PR, though.

Reply:
I have a PR coming later in the week that addresses this.  There is another set of JIRA's about removing unnecessary privileges and also working in a "low privilege" environment and I used those as an opportunity to clean-up this code.

Reply:
You'll need to adjust the code in this PR if you do.